modal/file_pattern_matcher.py

```diff
@@ -97,6 +112,9 @@ def _matches(self, file_path: str) -> bool:
         library. The reason is that `Matches()` in the original library is
         deprecated due to buggy behavior.
         """
+        if self._delayed_init:
```

Comment:
This is quite minor, but after thinking about it agian, I have a slight preference for putting this in from_file instead, making it clear that it's the "constructor's" responsibility to call init. Also, I'd put the `uninitialized._delayed_init = None` inside of this condition rather than in the delayed constructor function itself.

Reply:
discussed offline but we can't move it to `from_file` as that will read the file eagerly and our Image methods require it to be read lazily. Instead moved it to the `__call__` method